kratos/includes/amatrix_interface.h

```diff
@@ -609,6 +609,19 @@ typename TExpression1Type::data_type inner_prod(
 
 }
 
+    template <typename TExpressionType, std::size_t TCategory>
+    typename TExpressionType::data_type norm_1(
```

Comment:
same
also use `std::abs` instead of `std::fabs`

Reply:
why? std::abs works with only int and long